test/gtest/ucm/cuda_hooks.cc

```diff
@@ -115,14 +78,90 @@ class cuda_hooks : public ucs::test {
             void *ptr, size_t size,
             ucs_memory_type_t expect_mem_type = UCS_MEMORY_TYPE_CUDA)
     {
-        ASSERT_EQ(ptr, alloc_event.address);
-        ASSERT_GE(alloc_event.size, size);
-        EXPECT_TRUE((alloc_event.mem_type == expect_mem_type) ||
-                    (alloc_event.mem_type == UCS_MEMORY_TYPE_UNKNOWN));
+        check_event_present(m_alloc_events, "alloc", ptr, size,
+                            expect_mem_type);
+    }
+
+    void check_mem_free_events(void *ptr, size_t size)
```

Comment:
```suggestion
    void check_mem_free_events(void *ptr, size_t size) const
```
As well as `check_mem_alloc_events`.